source/code/providers/Container_HostInventory_Class_Provider.cpp

```diff
@@ -0,0 +1,212 @@
+/* @migen@ */
+#include <MI.h>
+#include "Container_HostInventory_Class_Provider.h"
+
+#include <vector>
+#include <string>
+#include <syslog.h>
+#include <stdio.h>
+#include <uuid/uuid.h>
+
+#include "../cjson/cJSON.h"
+#include "../dockerapi/DockerRemoteApi.h"
+#include "../dockerapi/DockerRestHelper.h"
+
+using namespace std;
+
+MI_BEGIN_NAMESPACE
+
+class Guid
+{
+public:
+    ///
+    /// Create a guid and represent it as string
+    ///
+    static string NewToString()
+    {
+        uuid_t uuid;
+        uuid_generate_random(uuid);
+        char s[37];
+        uuid_unparse(uuid, s);
+        return string(s);
+    }
+};
+
+class ContainerHostInventoryQuery
+{
+public:
+    ///
+    /// Get logs for all containers on the host
+    ///
+    /// \returns Object representing the host
+    ///
+    static Container_HostInventory_Class InspectHost()
+    {
+        Container_HostInventory_Class hostInventoryInstance;
+
+        // Request containers
+        vector<string> request(1, DockerRestHelper::restDockerInfo());
+        vector<cJSON*> response = getResponse(request);
+
+        // See https://docs.docker.com/engine/api/v1.27/#section/Versioning for example output
```

Comment:
Not a particular version. GET /info HTTP/1.1\r\nHost: localhost\r\n\r\n .  Since things are still in flux maybe we can lock the version down the line. I would like to be flexible at this point.